selfdrive/car/subaru/carstate.py

```diff
@@ -1,10 +1,36 @@
+from collections import deque
 import copy
 from cereal import car
 from opendbc.can.can_define import CANDefine
 from common.conversions import Conversions as CV
 from selfdrive.car.interfaces import CarStateBase
 from opendbc.can.parser import CANParser
-from selfdrive.car.subaru.values import DBC, CAR, GLOBAL_GEN2, PREGLOBAL_CARS, SubaruFlags
+from selfdrive.car.subaru.values import DBC, CAR, GLOBAL_GEN2, PREGLOBAL_CARS, Buttons, SubaruFlags
+
+PREV_BUTTON_SAMPLES = 8
+
+def cruise_buttons_conversion(cruise_buttons):
```

Comment:
Think the other ports use a dict constant in values.py or similar

Reply:
if we merge "Main", "Resume", and "Set" into a single DBC field then we could do that (like preglobal has right now)